pkg/kubelego/kubelego.go

```diff
@@ -28,6 +28,20 @@ import (
 var _ kubelego.KubeLego = &KubeLego{}
 
 func makeLog() *log.Entry {
+	logtype := strings.ToLower(os.Getenv("LEGO_LOG_TYPE"))
+	if logtype == "" {
+		logtype = "text"
+	}
+
+	if logtype == "json" {
+		log.SetFormatter(&log.JSONFormatter{})
+	} else if logtype == "text" {
+		log.SetFormatter(&log.TextFormatter{})
+	} else {
+		log.WithField("logtype", logtype).Fatal("Given logtype was not valid, check LEGO_LOG_TYPE configuration")
+		os.Exit(1)
```

Comment:
Not one for this PR, but in future I'd like to refactor this method out of here as we shouldn't really be calling `os.Exit` here. Ideally this method would exist elsewhere, and the `New` function should accept some form of `logger` parameter.